flink-runtime/src/main/java/org/apache/flink/runtime/state/CheckpointStorageWorkerView.java

```diff
@@ -75,4 +77,14 @@ CheckpointStreamFactory resolveCheckpointStorageLocation(
      * @return A toolset for additional operations for state owned by tasks.
      */
     CheckpointStateToolset createTaskOwnedCheckpointStateToolset();
+
+    /**
+     * Return {@link org.apache.flink.runtime.state.filesystem.FsMergingCheckpointStorageAccess} if
+     * file merging is enabled. Otherwise, return itself.
+     */
+    default CheckpointStorageWorkerView toFileMergingStorage(
```

Comment:
It might be a little weird to have the interface moved here in ```CheckpointStorageWorkerView```. The interface does not make much sense other than in ```FsCheckpointStorageAccess``` (like in ```MemoryBackendCheckpointStorageAccess```).

Maybe we can at least make it clearer in the comments here: 'Return xxx if file merging is enabled and supported by this ```CheckpointStorageWorkerView```'

Reply:
Thanks for your suggestion, putting `toFileMergingStorage` into `CheckpointStorageWorkerView` is to reduce the [type checking in StreamTask](https://github.com/apache/flink/pull/22890#discussion_r1251648094).